metals/src/main/scala/scala/meta/internal/metals/codeactions/ExtractValueCodeAction.scala

```diff
@@ -0,0 +1,311 @@
+    textEdits match {
+      case Some(edits) =>
+        val codeAction = new l.CodeAction()
+        codeAction.setTitle(ExtractValueCodeAction.title)
+        codeAction.setKind(this.kind)
+        codeAction.setEdit(
+          new l.WorkspaceEdit(
+            Map(path.toURI.toString -> edits.asJava).asJava
+          )
+        )
+        Seq(codeAction)
+      case None => Nil
+    }
+  }
+
+  /**
+   * If statement's direct parent is Def it means it's a
+   * single line method and we need to add braces.
+   *
+   * @param stat statement we are extracting from
+   * @param source full text of the file
+   * @param valueString extracted value to add
+   * @param blank whietespace character to use for indentation
+   * @return text edits together with braces to add
+   */
+  private def withBraces(
+      stat: Tree,
+      source: String,
+      valueString: String,
+      blank: Char
+  ): Seq[l.TextEdit] = {
+
+    def defnEqualsPos(defn: Defn.Def): Option[Position] = defn.tokens.reverse
+      .collectFirst {
+        case t: Token.Equals if t.start < defn.body.pos.start => t.pos
+      }
+
+    val edits = for {
+      defn <- stat.parent.collect { case defn: Defn.Def => defn }
+      equalsPos <- defnEqualsPos(defn)
+      defnLineIndentation = blank.stringRepeat(
+        indentationLength(source, defn.pos.start - defn.pos.startColumn)
+      )
+      additionalIndent =
+        if (defnLineIndentation.headOption.exists(_ == '\t')) "\t"
+        else "  "
+      innerIndentation = defnLineIndentation + additionalIndent
+    } yield {
+      val statStart = stat.pos.toLSP
+      statStart.setEnd(statStart.getStart())
+
+      val startBlockPos = equalsPos.toLSP
+      startBlockPos.setStart(startBlockPos.getEnd())
+      startBlockPos.setEnd(statStart.getStart())
+
+      val noIndentation = equalsPos.startLine == defn.body.pos.startLine
+
+      // Scala 3 optional braces
+      if (canUseBracelessSyntax(stat, source)) {
+        // we need to create a new indented region
+        if (noIndentation) {
+          Seq(
+            new l.TextEdit(
+              startBlockPos,
+              s"""|
+                  |$additionalIndent$valueString
+                  |$innerIndentation""".stripMargin
+            )
+          )
+        }
+        // make sure existing indentation after `=` is correct
+        else {
+          val statIndentation =
+            indentationLength(source, stat.pos.start - stat.pos.startColumn)
+          val statAdditionalIndentation =
+            if (statIndentation <= defnLineIndentation.size)
+              (defnLineIndentation.size - statIndentation) + additionalIndent.size
+            else 0
+          val indentStat = blank.stringRepeat(statAdditionalIndentation)
+
+          Seq(
+            new l.TextEdit(
+              statStart,
+              s"""|$indentStat${valueString.trim()}
+                  |$innerIndentation""".stripMargin
+            )
+          )
+        }
+        // Scala 2
```

Comment:
Just to make sure I'm tracking this is actually 
```suggestion
        // Scala 2 and Scala 3 non signification whitespace syntax
```
right?